gcsfs/core.py

```diff
@@ -389,7 +554,6 @@ def mkdir(self, bucket, acl='projectPrivate',
                    predefinedDefaultObjectAcl=default_acl,
                    json={"name": bucket})
         self.invalidate_cache(bucket)
-        self.invalidate_cache('')
 
     def rmdir(self, bucket):
         """Delete an empty bucket"""
```

Comment:
I wonder, if you delete the last key within a given prefix, which calls invalidate cache on the parent, do we expect the apparent directory to disappear?
e.g.,

gcs.ls('bucket/')
['bucket/thing/']
gcs.ls('bucket/thing/')
['bucket/thing/key']
gcs.rm('bucket/thing/key')
gcs.ls('bucket/')
[]  # directory should be gone